File: Day16/Output/optimized.c

#include <stdio.h>
#include <time.h>

double calculate(int iterations, double param1, double param2) {
    double result = 1.0;
    for (int i = 1; i <= iterations; i++) {
        double j = i * param1 - param2;
        result -= (1.0 / j);
        j = i * param1 + param2;
        result += (1.0 / j);
    }
    return result;
}

int main() {
    clock_t start_time = clock();
    
    double result = calculate(100000000, 4.0, 1.0) * 4.0;
    
    clock_t end_time = clock();
    double execution_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

    printf("Result: %.12f\n", result);
    printf("Execution Time: %.6f seconds\n", execution_time);
    
    return 0;
}

File: Day16/Output/optimized.swift

import Foundation

func calculate(iterations: Int, param1: Double, param2: Double) -> Double {
    var result = 1.0
    for i in 1...iterations {
        let j1 = Double(i) * param1 - param2
        let j2 = Double(i) * param1 + param2
        result -= 1.0 / j1
        result += 1.0 / j2
    }
    return result
}

let startTime = CFAbsoluteTimeGetCurrent()
let result = calculate(iterations: 100_000_000, param1: 4.0, param2: 1.0) * 4.0
let endTime = CFAbsoluteTimeGetCurrent()

print(String(format: "Result: %.12f", result))
print(String(format: "Execution Time: %.6f seconds", endTime - startTime))
